django/db/models/sql/query.py

```diff
@@ -65,20 +65,58 @@ def get_field_names_from_opts(opts):
     )
 
 
+def get_paths_from_expression(expr):
+    if isinstance(expr, F):
+        yield expr.name
+    elif hasattr(expr, "flatten"):
+        for child in expr.flatten():
+            if isinstance(child, F):
+                yield child.name
+            elif isinstance(child, Q):
+                yield from get_children_from_q(child)
+
+
 def get_children_from_q(q):
     for child in q.children:
         if isinstance(child, Node):
             yield from get_children_from_q(child)
-        else:
-            yield child
+        elif isinstance(child, tuple):
+            lhs, rhs = child
+            yield lhs
+            if hasattr(rhs, "resolve_expression"):
+                yield from get_paths_from_expression(rhs)
+        elif hasattr(child, "resolve_expression"):
+            yield from get_paths_from_expression(child)
 
 
-def get_child_with_renamed_prefix(prefix, replacement, child):
+def replace_child_name(child_name, prefix, replacement):
+    return child_name.replace(prefix, replacement, 1)
+
+
+def get_child_with_renamed_prefix(prefix, replacement, child, is_lhs=True):
     if isinstance(child, Node):
         return rename_prefix_from_q(prefix, replacement, child)
-    lhs, rhs = child
-    lhs = lhs.replace(prefix, replacement, 1)
-    return lhs, rhs
+    if isinstance(child, tuple):
+        lhs, rhs = child
+        lhs = replace_child_name(lhs, prefix, replacement)
+        if hasattr(rhs, "resolve_expression"):
+            rhs = get_child_with_renamed_prefix(prefix, replacement, rhs, is_lhs=False)
```

Comment:
What do you think about skipping `F` instances instead of adding an extra kwarg?, e.g.
```diff
diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
index 2634a33d98..1b476a6dcf 100644
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -93,19 +93,18 @@ def replace_child_name(child_name, prefix, replacement):
     return child_name.replace(prefix, replacement, 1)
 
 
-def get_child_with_renamed_prefix(prefix, replacement, child, is_lhs=True):
+def get_child_with_renamed_prefix(prefix, replacement, child):
     if isinstance(child, Node):
         return rename_prefix_from_q(prefix, replacement, child)
     if isinstance(child, tuple):
         lhs, rhs = child
         lhs = replace_child_name(lhs, prefix, replacement)
-        if hasattr(rhs, "resolve_expression"):
-            rhs = get_child_with_renamed_prefix(prefix, replacement, rhs, is_lhs=False)
+        if not isinstance(rhs, F) and hasattr(rhs, "resolve_expression"):
+            rhs = get_child_with_renamed_prefix(prefix, replacement, rhs)
         return lhs, rhs
     if isinstance(child, F):
         child = child.copy()
-        if is_lhs:
-            child.name = replace_child_name(child.name, prefix, replacement)
+        child.name = replace_child_name(child.name, prefix, replacement)
         return child
     if hasattr(child, "resolve_expression"):
         child = child.copy()

```